addons/project/project.py

```diff
@@ -396,11 +358,14 @@ def onchange_planned(self, cr, uid, ids, planned=0.0, effective=0.0):
         return {'value': {'remaining_hours': planned - effective}}
 
     def onchange_project(self, cr, uid, id, project_id, context=None):
+        vals = {'partner_id': False}
         if project_id:
             project = self.pool.get('project.project').browse(cr, uid, project_id, context=context)
-            if project and project.partner_id:
-                return {'value': {'partner_id': project.partner_id.id}}
-        return {'value': {'partner_id': False}}
+            vals.update({
+                'project_type': project and project.project_type,
+                'partner_id': project and project.partner_id.id
```

Comment:
no need to check project.partner_id condition ? if we don't have project.partner_id then ?